app/labor/assign_tag_moderator.rb

```diff
@@ -10,28 +10,34 @@ def self.add_trusted_role(user)
     NotifyMailer.trusted_role_email(user).deliver
   end
 
-  def self.add_to_chat_channel(user)
-    ChatChannel.find_by(slug: "tag-moderators").add_users(user) if user.chat_channels.find_by(slug: "tag-moderators").blank?
-  end
-
-  def self.add_tag_mod_role(user, tag)
-    user.update(email_tag_mod_newsletter: true) if user.email_tag_mod_newsletter == false
-    user.add_role(:tag_moderator, tag)
-    Rails.cache.delete("user-#{user.id}/tag_moderators_list")
-    MailchimpBot.new(user).manage_tag_moderator_list
-  end
-
   def self.add_tag_moderators(user_ids, tag_ids)
     user_ids.each_with_index do |user_id, index|
       user = User.find(user_id)
       tag = Tag.find(tag_ids[index])
       add_tag_mod_role(user, tag)
       add_trusted_role(user)
-      add_to_chat_channel(user)
+      add_to_chat_channels(user, tag)
       NotifyMailer.tag_moderator_confirmation_email(user, tag.name).deliver unless tag.name == "go"
     end
   end
 
+  def self.add_to_chat_channels(user, tag)
+    ChatChannel.find_by(slug: "tag-moderators").add_users(user) if user.chat_channels.find_by(slug: "tag-moderators").blank?
```

Comment:
This line uses `find_by(...).blank?` whereas two lines below it's `where(...).none?` Maybe we should go for consistency here and use the same approach in both lines? FWIW the `where().none?` version produces the more efficient SQL query.